src/shared/case-editor/case-edit-page.component.ts

```diff
@@ -116,15 +118,41 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
       let caseEventData: CaseEventData = this.formValueService.sanitise(currentPageFields) as CaseEventData;
       caseEventData.event_token = this.eventTrigger.event_token;
       caseEventData.ignore_warning = this.ignoreWarning;
-      this.caseEdit.validate(caseEventData)
-        .subscribe(() => {
+      this.caseEdit.validate(caseEventData, this.currentPage.id)
+        .subscribe((jsonData) => {
+          this.updateFormData(jsonData as CaseEventData);
           this.saveDraft();
           this.next();
         }, error => this.handleError(error));
       this.scrollToTop();
     }
   }
 
+  updateFormData(jsonData: CaseEventData): void {
+    for (const caseFieldId of Object.keys(jsonData.data)) {
+      let wizardPage: WizardPage = this.wizard.findWizardPage(caseFieldId);
+      if (wizardPage) {
```

Comment:
Could a comment be added here to explain why we check for a wizard page? It doesn't seem obvious.

Reply:
Yes, added a comment and extracted the if condition.